scipy/ndimage/_filters.py

```diff
@@ -1989,7 +1989,8 @@ def _rank_filter(input, rank, size=None, footprint=None, output=None,
                 "A sequence of modes is not supported by non-separable rank "
                 "filters")
         mode = _ni_support._extend_mode_to_code(mode, is_filter=True)
-        if input.ndim == 1:
+        lim2 = input.size - ((footprint.size - 1) // 2 - origin)
+        if input.ndim == 1 and ((lim2 >= 0) or (input.size == 1)):
```

Comment:
So the fix is that for the corner case of `footprint` being large compared to array size, we're skipping this branch and therefore hitting the old `_nd_image.rank_filter` C implementation rather than the new `_rank_filter_1d.rank_filter` C++ code?

I think that deserves a comment here, because future readers will wonder why the 1-D specific implementation isn't being used for `input.ndim == 1`.

Reply:
I've added a comment similar to that, which also refers the reader back to this PR, and suggests that it may be possible to restore the fast path more broadly with algorithmic fixes.

That said, the original fast path author has clearly indicated they don't have the bandwidth to help us maintain it. Anyway, that's clearly why we're going through this. I don't know if it has been worth it to try to keep that fast path, and what percentage of real world cases have already been sent to the slower path as a result of working around problems, so things are getting a bit murky here, but the regression tests are the one thing that makes me think we're moving forward either way, because they were obviously missing before.